### PR TITLE
fix: Navigating from Recent Objects breadcrumb correctly updates the URL hash

### LINKED ISSUE #6151: Navigating to an object from a Breadcrumb doesn't update the URL hash
#### Summary

When navigating to an object via a Location Breadcrumb such as those in the Grand Search's search results or in Recent Objects, the URL hash is not updated. This will block e2e tests in certain situations if we need to wait for a URL update to confirm navigation has completed. This will also cause the tree to fail to highlight the focused item as the current location does not change.

#### Expected vs Current Behavior

The URL hash should always reflect the currently focused object

#### Steps to Reproduce

1. Navigate to an object via the Location  Breadcrumb either in the GrandSearch results or in Recent Objects
2. Notice that the currently focused object changes in the object view
3. Notice that the URL hash doesn't change and no longer reflects the focused object

#### Environment

* Open MCT Version:  `2.1.6-SNAPSHOT`
* Deployment Type:  local
* OS: MacOS
* Browser: Chrome

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [x] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

#### Additional Information

https://user-images.githubusercontent.com/9259993/213790951-fd2c4a2f-3297-4a2b-8923-ae1353794bcc.mov

### PR DESCRIPTION
**Pull request by ozyx**
Closes #6151  

### Describe your changes:

Provides a `navigateToPath` to `ObjectLabel` so that navigation from a breadcrumb works correctly. Previously, the navigationPath generated by the `ObjectLink` mixin was also including params, which navigates correctly but sometimes fails to update the URL hash due to logic within the Router. Params are set after navigation, and unless they're being changed (or we're opening a link in a new tab), there's no reason to include these in the navigation path.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by unlikelyzero — APPROVED** [review]
Looks great! Two questions.

Also, could you add a test stub to flex the max number of recent objects+1 to trigger a roll over?

**Comment by scottbell** [comment]
Updates hash as expected with both search results and recent objects:

https://user-images.githubusercontent.com/9853862/216012314-06d05aed-33da-4f22-bcdf-23cc0d9540a3.mov

File: e2e/tests/functional/recentObjects.e2e.spec.js [hunk 1]
```diff
@@ -79,7 +91,42 @@
...
+        await page.getByRole('navigation', {
+            name: `${clock.name} Breadcrumb`
+        }).locator('a').filter({
+            hasText: myItemsFolderName
+        }).click();
+
+        // Verify that the hash URL updates correctly
+        await waitForMyItemsNavigation;
+        // eslint-disable-next-line no-useless-escape
+        expect(page.url()).toMatch(new RegExp(`.*mine\?.*`));
```
[the remark this answers is not in the document]
Comment (scottbell) [reply e2e/tests/functional/recentObjects.e2e.spec.js:126]: Same as above

File: e2e/tests/functional/recentObjects.e2e.spec.js [hunk 2]
```diff
@@ -79,7 +91,42 @@ test.describe('Recent Objects', () => {
...
+        // Verify that the hash URL updates correctly
+        await waitForFolderNavigation;
+        // eslint-disable-next-line no-useless-escape
+        expect(page.url()).toMatch(new RegExp(`.*${folderA.uuid}\?.*`));
+
+        // Navigate to My Items by clicking on its entry in the Clock's breadcrumb
+        const waitForMyItemsNavigation = page.waitForURL(`**/mine?*`);
+        await page.getByRole('navigation', {
+            name: `${clock.name} Breadcrumb`
+        }).locator('a').filter({
```
[the remark this answers is not in the document]
Comment (scottbell) [reply e2e/tests/functional/recentObjects.e2e.spec.js:119]: image
Yeah, anchor element

**Review by scottbell — APPROVED** [review]
LGTM

**Pull request merged**